### PR TITLE
Tar ↔ SSH ↔ Docker Volume

### LINKED ISSUE #66: Server Side SFTP Handling
## User Story
As a Daemon, I would like to handle SFTP input from the client so that client-Daemon file sharing is enabled, and send files back.

## Detailed Description
Given the SSH connection made functional in #55 , add the ability to receive files from the client via SFTP.

## Acceptance Criteria
- [ ] Given SFTP input from client, then the server accepts input files
- [ ] Given output files, then the server send them to client via SFTP.
- [ ] Integration tests included

Points:
3

### PR DESCRIPTION
**Pull request by rvighne**
- New message formats for requests to the Daemon, which allow for sending or receiving files.
- `tar` support in the Daemon to convert from SSH streams to volume directory structure and vice versa.
- Use Docker volumes so the Makefile has access to the files. Make actually works now!
- Integration tests! The volume support is fully tested.
- Windows support. Only one integration test fails on Windows (not related to this PR).

Closes #66

### CODE REVIEW

File: test/unit/TempVolume.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	it.todo('removes the tempdir and its contents')
+	it.todo('reports failure if the directory cannot be removed')
+})
+
+describe('withTempDir', () => {
+	it.todo('creates the tempdir before the action is triggered')
+	it.todo('removes the tempdir after the action completes')
+	it.todo('removes the tempdir even when the action throws')
+	it.todo('propagates the promise status of the action')
+})
```
Comment (alexzma) [inline test/unit/TempVolume.test.ts:25]: May want to write this eventually.

**Review by alexzma — APPROVED** [review]
Looks good to me.

File: test/unit/TempVolume.test.ts (hunk 1 above)
Comment (rvighne) [reply test/unit/TempVolume.test.ts:25]: Yeah, these are pretty important. Unfortunately I have literally no time to do so...

Anyone who wants to write some testcases is free to tackle these if they want 😇

**Review by pjot726 — APPROVED** [review]
I have read the code in files relevant to the SSH Client implementation and I think I have a good grasp of them. Things look good on that front, I think @alexzma will have a better understanding of the Daemon-Docker code.

**Pull request merged**